cmake/detect_cxx_version.cmake

```diff
@@ -0,0 +1,53 @@
+set(cxx17_minimum_msvc_version 19.14)
```

Comment:
Why do you need these checks while you already do check at the C++ level in the `include/frozen/bits/defines.h` file?

Reply:
This for CMake to activate C++17/20 at configure time for the tests. The defines are there for the code to detect whether C++17 has been enabled, either by us in the tests or in the user's project.

Reply:
I just noticed that some variables at the end are not needed anymore, I'll remove them.